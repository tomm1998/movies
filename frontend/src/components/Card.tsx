import React from "react";

export const Card = ({
  header,
  children,
  actionLabel,
  handleAction,
  image,
  imagePosition = "center",
}) => {
  return (
    <div
      style={{
        padding: "16px",
        borderRadius: "8px",
        width: "100%",
        boxShadow:
          "-60px 0px 100px -90px #000000, 60px 0px 100px -90px #000000",
      }}
    >
      {image && imagePosition === "top" && <img src={image} />}
      {header ||
        (actionLabel && (
          <div>
            {header && <h1>{header}</h1>}
            {actionLabel && (
              <button onClick={handleAction}>{actionLabel}</button>
            )}
          </div>
        ))}
      {image && imagePosition === "center" && <img src={image} />}
      <div>{children}</div>
      {image && imagePosition === "bottom" && <img src={image} />}
    </div>
  );
};
