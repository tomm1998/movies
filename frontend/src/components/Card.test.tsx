import { render } from "@testing-library/react";
import React from "react";
import { describe, it, expect } from "vitest";
import Card from "./Card";

describe("Card", () => {
  it("renders", () => {
    const wrapper = render(<Card children="child" />);
    expect(wrapper.container).toMatchSnapshot();
  });
});
