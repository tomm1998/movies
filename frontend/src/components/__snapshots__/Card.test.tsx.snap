// Vitest Snapshot v1

exports[`Card > renders 1`] = `
<div>
  <div
    style="padding: 16px; border-radius: 8px; width: 100%; box-shadow: -60px 0px 100px -90px #000000, 60px 0px 100px -90px #000000;"
  >
    <div>
      child
    </div>
  </div>
</div>
`;
